docs/source/configuration.rst

```diff
@@ -19,13 +19,44 @@ any vales read from earlier files.
 - :code:`tox.ini`
 - :code:`pep8.ini`
 - :code:`.sqlfluff`
+- :code:`pyproject.toml`
 
-Within these files, they will be read like an `cfg file`_, and *SQLFluff*
-will look for sections which start with *SQLFluff*, and where subsections
-are delimited by a semicolon. For example the *jinjacontext* section will
-be indicated in the section started with *[sqlfluff:jinjacontext]*.
+Within these files, the first four will be read like an `cfg file`_, and
+*SQLFluff* will look for sections which start with *SQLFluff*, and where
+subsections are delimited by a semicolon. For example the *jinjacontext*
+section will be indicated in the section started with
+*[sqlfluff:jinjacontext]*.
+
+For the `pyproject.toml file`_, all valid sections start with `tool.sqlfluff`
+and subsections are delimited by a dot and suffixed with *__options*.
```

Comment:
You mentioned the `__options` suffix in the PR description. Can you give an example of the kind of conflict you're avoiding by doing this?

If I understand correctly, could the toml reader skip sections that don't start with `tools.sqlfluff` or `sqlfluff`?

Reply:
So a TOML is basically a nested  object/dictionary so the keys would collide. The following code raises an error

```python
import toml

content = """
[tool.sqlfluff]
templater = "jinja"

[tool.sqlfluff.templater]
unwrap_wrapped_queries = true
"""

toml.loads(content)
```

The error is:

```
What? templater already exists?{'templater': 'jinja'} (line 5 column 1 char 38)
```

> If I understand correctly, could the toml reader skip sections that don't start with tools.sqlfluff or sqlfluff?

Yes, the other sections are ignored. The convention I believe is to put every set of configurations for `my_tool` under `[tool.my_tool*]` sections.